src/main/java/com/minecolonies/coremod/blocks/BlockDecorationController.java

```diff
@@ -114,6 +125,30 @@ public InteractionResult use(
         return InteractionResult.SUCCESS;
     }
 
+    @Override
+    public void setPlacedBy(@NotNull final Level worldIn, @NotNull final BlockPos pos, final BlockState state, final LivingEntity placer, final ItemStack stack)
+    {
+        super.setPlacedBy(worldIn, pos, state, placer, stack);
+
+        /*
+        Only work on server side
+        */
+        if (worldIn.isClientSide)
+        {
+            return;
+        }
+
+        final BlockEntity tileEntity = worldIn.getBlockEntity(pos);
+        if (tileEntity instanceof TileEntityDecorationController && ((TileEntityDecorationController) tileEntity).getPositionedTags().getOrDefault(BlockPos.ZERO, new ArrayList<>()).contains(LEISURE))
+        {
+            @Nullable final IColony colony = IColonyManager.getInstance().getColonyByPosFromWorld(worldIn, pos);
+            if (colony != null)
+            {
+                colony.getBuildingManager().addLeisureSite(pos);
```

Comment:
See the note on `IBuildingManager` this is not directly related to a building, so move it to a separate manager to clean up the API.